[release-1.5] RHIDP-4741: RHDH 1.5 Release Notes (#979)
redhat-developer/red-hat-developers-documentation-rhdh: modules/release-notes/ref-release-notes-new-features.adoc

This section highlights new features in {product} {product-version}.

[id="feature-rhidp-5499"]

> == `techdocsAddons` configuration for dynamic plugins

Reviewer: ```suggestion
== Added `techdocsAddons` configuration for dynamic plugins
```

 frontend:
   my-plugin-package:
     signInPage:
       importName: CustomSignInPage
----

** The exported `CustomSignInPage` will be mapped to `components.SignInPage` during front-end initialization.
--

* Authentication provider settings:
+
--
** A new `providerSettings` configuration field allows front-end dynamic plugins to define authentication provider settings for display in the *Authentication Providers* tab on the *Settings* page.
+
.Example configuration
[source,yaml]
----
dynamicPlugins:
 frontend:
   my-plugin-package:
     providerSettings:
- title: Github Two
description: Sign in with GitHub Org Two
provider: core.auth.github-two
----

** Each `providerSettings` entry creates a corresponding row in the *Authentication Providers* tab. The provider field must match the string used in `createApiRef` when defining the API reference:
+
.Example configuration
[source,javascript]
----
export const ghTwoAuthApiRef: ApiRef<
 OAuthApi & ProfileInfoApi & BackstageIdentityApi & SessionApi
> = createApiRef({
 id: 'core.auth.github-two', // <--- this string
})
----
--

For more information, see <doc links>.

[id="feature-rhidp-5513"]
== Floating action button support for plugins

> {product-very-short} now supports adding plugins as floating action on selected pages. This feature allows you to configure commonly used plugins as floating buttons, providing quick access and improving navigation by centralizing essential plugins on a single page.

Reviewer 2: ```suggestion
{product-very-short} now supports adding plugins or external links as floating actions on {product}. This feature allows you to configure commonly used plugins as floating buttons, providing quick access and improving navigation by centralizing essential plugins on a single page.
```

== Improved status conditions for Backstage custom resources

This update enhances the status conditions in the Backstage custom resource(s) managed by the {product-short} Operator, providing clearer insights into application availability.

A new `DeployInProgress` reason has been introduced under the `Deployed` condition. It appears when the application is still starting up and not fully available. The `Deployed` reason now only reflects when the {product-short} application is fully up and running with the desired number of replicas, improving visibility during deployment. For example:

.Example status conditions
[source,terminal,subs="+quotes,+attributes"]
----
Status:
  Conditions:
    Last Transition Time: 2025-03-10T17:09:22Z
    Message: Deployment status: Available: Deployment does not have minimum availability. Progressing: ReplicaSet "backstage-bs1-674476fd67" is progressing.
    Reason: DeployInProgress
    Status: False
    Type: Deployed

...
Status:
  Conditions:
    Last Transition Time: 2025-03-10T17:14:43Z
    Message:
    Reason: Deployed
    Status: True
    Type: Deployed
----

[id="enhancement-rhidp-4785"]
== Enhanced secret configuration for YAML files

> {product-very-short} now introduces the ability to configure multiple secrets within a single YAML file, similar to PVCs. Additionally, it allows annotations to specify:

Reviewer 3: ```suggestion
The {product-very-short} Operator now introduces the ability to configure multiple secrets within a single YAML file, similar to PVCs. Additionally, it allows annotations to specify:
```

This relates to the Operator.

Author: Done
